client/components/forms/form-text-input/index.jsx

```diff
@@ -17,24 +14,28 @@ export default class FormTextInput extends PureComponent {
 		className: PropTypes.string,
 	};
 
-	constructor() {
-		super( ...arguments );
+	currentTextField = undefined;
 
-		this.selectOnFocus = this.selectOnFocus.bind( this );
-	}
+	textFieldRef = element => {
+		if ( element && this.props.inputRef ) {
+			this.props.inputRef.current = element;
+		}
+		this.currentTextField = element;
+	};
```

Comment:
The `inputRef` can also be a function that's supposed to be called. You want to emulate the React ref-setting code that's [here](https://github.com/facebook/react/blob/d9b4c55d53c67045c38b12f5aad57a163e7f63a2/packages/react-reconciler/src/ReactFiberCommitWork.js#L660-L701).

Reply:
Done, thanks! I'm silently ignoring `inputRef`s that don't match one of the two shapes, as I don't think there's the need to throw an exception here.

Reply:
If I pass some invalid object as a `ref` and it will be ignored, I'll have a very hard time debugging what's wrong. My `ref`s are not set and there are no errors! It might be better letting the `ref.current` assignment fail. React internally does that, it just emits an informative warning in dev mode.

Reply:
That sounds reasonable. Reworked it to make it fail now.